python/lib/core/dmod/core/dataset.py

```diff
@@ -177,19 +177,42 @@ def __init__(self, manager: DatasetManager = None, **kwargs):
                 raise ValueError(f"Expected UUID got {type(manager.uuid)}")
             self.manager_uuid = manager.uuid
 
+    def __eq__(self, other):
+        return (
+            isinstance(other, Dataset)
+            and self.access_location == other.access_location
+            and self.category == other.category
+            and self.created_on == other.created_on
+            and self.data_domain == other.data_domain
+            and self.dataset_type == other.dataset_type
+            and self.is_read_only == other.is_read_only
+            and self.name == other.name
+        )
+
     @property
     def manager(self) -> Optional[DatasetManager]:
+        """
+        The ::class:`DatasetManager` for this instance.
+
+        Returns
+        -------
+        DatasetManager
+            The ::class:`DatasetManager` for this instance.
+        """
         return self._manager
 
-    @manager.setter
-    def manager(self, value: DatasetManager = None):
-        self._manager = value if isinstance(value, DatasetManager) else None
+    def set_manager(self, value: DatasetManager):
+        """
+        Sets the ::class:`DatasetManager` and updates the
+        ::attribute:`manager_uuid` property on this instances.
 
-        if value is not None:
-            # pydantic will not validate this, so we need to check it
-            if not isinstance(value.uuid, UUID):
-                raise ValueError(f"Expected UUID got {type(value.uuid)}")
-            self.manager_uuid = value.uuid
+        Parameters
+        ----------
+        value: DatasetManager
+            The new value for ::attribute:`manager`.
+        """
+        self._manager = value
```

Comment:
This change removes support for unsetting the manager.  We weren't entirely handling the case properly, but I'm still curious why remove it rather than fix it.   And are we sure that we don't need it?

Also, why not also renaming the backing attribute to `manager`?

Reply:
Ah nice catch! I shouldn't have removed that capability. I was only thinking you would ever want to set a manager, but it makes sense that you may want to disassociate a manager too.

> Also, why not also renaming the backing attribute to `manager`?

If we change back to `manager`, if you were to set through the `manager` property, you wouldn't update the `manager_uuid` property unless you went through the `set_manager` method. My thinking has been for properties where in the past we'd done more than just set the properties value, hide the property and provide a setter.